### ISSUE TITLE
unassigned variables in upper case not detected 

### ISSUE THREAD

**Issue by deepesh1081** [koalaman/shellcheck#1037]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [- ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [- ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related #351 is there since 2015, looks it was not resolved.  

I just got this today and also check it on shellcheck.net same behavior:
 
shellcheck -V
ShellCheck - shell script analysis tool
version: 0.4.6
license: GNU General Public License, version 3
website: http://www.shellcheck.net

#### Here's a snippet or screenshot that shows the problem:
shellcheck /tmp/test

In /tmp/test line 2:
dbname=$car
       ^-- SC2154: car is referenced but not assigned.

In /tmp/test line 5:
echo $dbname
     ^-- SC2086: Double quote to prevent globbing and word splitting.

In /tmp/test line 6:
echo $DBNAME
     ^-- SC2086: Double quote to prevent globbing and word splitting.

#!/your/interpreter
your script here

```
cat  /tmp/test
#!/bin/bash
dbname=$car
DBNAME=$CAR

echo $dbname
echo $DBNAME

#### Here's what shellcheck currently says:
Shellcheck does not detects that an unassigned variable if it is in upper case 


#### Here's what I wanted or expected to see:
I was expecting it will call out that $CAR is not assigned

**Comment by koalaman** [comment]
At the moment this is intentional: uppercase is conventionally used for environment variables that will be inherited from the parent.

Typical examples are `PATH`, `LANG`, `TERM`, `TZ`, `EDITOR`, `PAGER`, `JAVA_HOME`, `PYTHONPATH`, `SUDO_USER`, `SSH_CLIENT`, all of which are generally assumed to be set when relevant, rather than being defined by the current script.

There is currently no good story for how to define which uppercase variables this does and doesn't apply to.

**Comment by deepesh1081** [comment]
Sure, I switched to Upper case instead of UPPER case, to work around it. my only thought would be to may be print a warning stating something like "Unclear if $VAR is defined outside of this script or just undefined"

Thanks for reviewing this.

**Comment by nihilus** [comment]
I did a test today and changed the code from lowercase to use uppercase for this, I found 3-4 variables quick that was wrong and some hundreds of false positives. I didn't check them all, however having a switch for enabling this behaviour would be nice.

**Comment by Nightfirecat** [comment]
Perhaps this would be another good candidate for implementing once [the big verbosity feature](https://github.com/koalaman/shellcheck/issues/971#issuecomment-325967572) is in place?

**Comment by h0nIg** [comment]
any update for this? i'm interested in this as well

**Comment by koalaman** [comment]
No updates at this time.
